### ISSUE TITLE
Sort or  Group "text" channel results.

### ISSUE THREAD

**Issue by cozycode** [alexpasmantier/television#194]
**Problem**
When using the "text" channel to search within files across a directory tree, it can be challenging to make sense of the results because they appear in a random order.  Matches from each file are interleaved with results from other files and the line numbers within each file also appear in random order.

**Feature**
After entering a search phrase when using the "text" channel/mode, it would be nice if the result list would sort or group matches from within the same file. It might also be nice if the matches from the same file were ordered in the same order they appear in the file. Right now I am seeing matches from the same file interleaved with matches from other files.  

**Additional context**
It might also be nice to have a toggle to list each file with matches exactly once in the left side (with the match locations highlighted in the preview text on right) or to list each match as it does now.

**Comment by xosxos** [comment]
This would be a great feature. A good implementation of this could also allow the git log entries to appear in chronological order.

**Comment by zyoNoob** [comment]
Since you did mention this here, I will give whatever info I found while trying to fix this myself but gave up because of time constraints.

- FzF handles history in a unique way, I didn't study the exact approach but hey have separate flow for deciding the order of history, might be worth checking it out.
- Nucleo the library that TV uses for fuzzy matching currently doesn't supporting no-sort or sorted behavior as mentioned on their repository, but I did check some pull requests created by a contributor that does sort - TLDR direct sorting on time and score didn't work correctly, and top of that I didn't manage to integrate the call for sorted list in the TV code.
- I tried forking TV to do sorting on TV side of things but it doesnt work properly because TV fetches a batch of search results using nucleo and then renders them, so when ever i would scroll the results the search results would resort in the current pane. From what i figured is to reduce memory usage and compute overhead.

So long story short from my research, FzF uses a multi-condition sort for history and search on it, and for TV we might need feature additions to nucleo.

**Comment by raylu** [comment]
> fzf `^r` command history search is sorted by last-used initially, but once you start typing to search it appears to be sorted by string length

I don't think that's right. it uses `--scheme history`: https://github.com/junegunn/fzf/blob/84e515bd6ecff0ed07d1373ff8e770f91d933b2e/shell/key-bindings.zsh#L135
and you can try
```sh
echo "prefix 1\nprefix 22222\ncow goes moo\nprefix 3" | fzf --scheme history
```
and see that typing `prefix` keeps 22222 in the middle

**Comment by raylu** [comment]
https://github.com/helix-editor/nucleo/pull/72 implements what we need for history mode, I think. it was opened 6 months ago, landed 2 months ago without comment, but nucleo hasn't had a release in over a year

ugly hack to turn it on always
```diff
diff --git a/Cargo.toml b/Cargo.toml
index b6f0f3bbe6..0a183790ce 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -38,7 +38,7 @@
 signal-hook = "0.3"
 human-panic = "2.0"
 parking_lot = "0.12"
-nucleo = "0.5"
+nucleo = { git = "https://github.com/helix-editor/nucleo.git", rev = "5b74652e482f7c07d827f18c6d21e7540c242c69" }
 toml = "0.8"
 lazy-regex = { version = "3.4.1", features = [
   "lite",
diff --git a/television/matcher/mod.rs b/television/matcher/mod.rs
index b27d2f9f95..8536fbafe0 100644
--- a/television/matcher/mod.rs
+++ b/television/matcher/mod.rs
@@ -61,13 +61,15 @@
 {
     /// Create a new fuzzy matcher with the given configuration.
     pub fn new(config: &config::Config) -> Self {
+        let mut nucleo = nucleo::Nucleo::new(
+            config.into(),
+            Arc::new(|| {}),
+            config.n_threads,
+            1,
+        );
+        nucleo.sort_results(false);
         Self {
-            inner: nucleo::Nucleo::new(
-                config.into(),
-                Arc::new(|| {}),
-                config.n_threads,
-                1,
-            ),
+            inner: nucleo,
             total_item_count: 0,
             matched_item_count: 0,
             status: Status::default(),
```
and now
```sh
echo "prefix 1\nprefix 22222\ncow goes moo\nprefix 3" | cargo r
```
gives the behavior we want for history mode
